### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by binhussen**
## TDD project
In this PR, we have implemented:
- Created Tests for `factorial`, `reverse`, and `fizzbuzz` functions.
- Added a class called Solver.
- Created a method called `factorial`, `reverse`, and `fizzbuzz` within their functionalities.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  it 'should return fizzbuzz' do
+    Solver.new.fizzbuzz(30).should == 'fizzbuzz'
+  end
+
+  it 'should return number' do
+    Solver.new.fizzbuzz(7).should == (7)
+  end
+end
```
Comment (tresorsawasawa) [inline spec/solver_spec.rb]: - [ **OPTIONAL** ] For the **[DRY](https://www.techtarget.com/whatis/definition/DRY-principle)** principle, best practices and to make your code more descriptive, I suggest you update the selected code above to match the template below:
```ruby
describe Solver do
  describe '#factorial' do

    # All tests for the factorial method here
    
  end

  describe '#reverse' do

    # All tests for the reverse method here
    
  end

  describe '#fizzbuzz' do
    
    # All tests for the fizzBuzz method here

  end
end
```

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require './solver'
+
+describe Solver do
+  it 'should return 1 for 0' do
+    Solver.new.factorial(0).should == 1
+  end
+  it 'should return 120 for 5' do
+    Solver.new.factorial(5).should == 120
+  end
```
Comment (tresorsawasawa) [inline spec/solver_spec.rb]: - [ ] Kindly make sure the `#factorial` method has tests that handle negative value(s) which raises an exception as an expectation for the project requirements.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @binhussen and @edi-sipka 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 2 above)
Comment (binhussen) [reply spec/solver_spec.rb]: Thank you @tresorsawasawa for your feedback, I will add raises exception test.

**Review by Ismailco — APPROVED** [review]
### Hi @binhussen, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by binhussen** [comment]
Thank you, @Ismailco for your feedback and for approving our code

**Pull request merged**